File: 1653/1653.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        bs = 0
        minDeletions = 0
        for char in s:
            if char == 'a':
                minDeletions = min(minDeletions + 1, bs)
            else:
                bs += 1   
        return minDeletions

File: 1653/1653.ts
function minimumDeletions(s: string): number {
    let bs: number = 0;
    let count: number = 0;
    for (var i: number = 0; i < s.length; i++){
        if (s.charAt(i)=='a') count = Math.min(count+1, bs);
        else bs+=1;
    }

    return count;
};
